boards/arm/nrf52840_pca10059/doc/index.rst

```diff
@@ -107,82 +107,189 @@ Programming and Debugging
+These instructions use the :ref:`west <west>` tool and assume you are in the
+root directory of your :term:`west installation`.
 
-.. zephyr-app-commands::
-   :zephyr-app: samples/basic/blinky
-   :board: nrf52840_pca10059
-   :goals: build
+Option 1: Using the Built-In Bootloader Only
+--------------------------------------------
+
+The board is factory-programmed with Nordic's bootloader from Nordic's nRF5
+SDK. With this option, you'll use Nordic's `nrfutil`_ program to create
+firmware packages supported by this bootloader and flash them to the
+device. Make sure ``nrfutil`` is installed before proceeding.
+
+#. Reset the board into the Nordic bootloader by pressing the RESET button.
+
+   The push button is on the far side of the board from the USB connector. Note
+   that the button does not face up. You will have to push it from the outside
+   in, towards the USB connector:
+
+   .. image:: img/nRF52840_dongle_press_reset.svg
+      :alt: Location of RESET button and direction of push
+
+   The red LED should start a fade pattern, signalling the bootloader is
+   running.
+
+#. Compile a Zephyr application; we'll use :ref:`blinky <blinky-sample>`.
+
+   .. zephyr-app-commands::
+      :app: zephyr/samples/basic/blinky
+      :board: nrf52840_pca10059
+      :goals: build
+      :tool: west
+
+#. Package the application for the bootloader using ``nrfutil``:
+
+   .. code-block:: console
+
+      nrfutil pkg generate --hw-version 52 --sd-req=0x00 \
+              --application build/zephyr/zephyr.hex \
+              --application-version 1 blinky.zip
+
+#. Flash it onto the board. Note :file:`/dev/ttyACM0` is for Linux; it will be
+   something like ``COMx`` on Windows, and something else on macOS.
+
+   .. code-block:: console
+
+      nrfutil dfu usb-serial -pkg blinky.zip -p /dev/ttyACM0
+
+   When this command exits, observe the green LED on the board blinking,
+   instead of the red LED used by the bootloader.
+
+For more information, see `Nordic Semiconductor USB DFU`_.
+
+Option 2: Using MCUboot in Serial Recovery Mode
+-----------------------------------------------
+
+It is also possible to use the MCUboot bootloader with this board to flash
+Zephyr applications. You need to do some one-time set-up to build and flash
+MCUboot on your board. From that point on, you can build and flash other Zephyr
+applications using MCUboot's serial recovery mode. This process does not
+overwrite the built-in Nordic bootloader, so you can always go back to using
+Option 1 later.
+
+Install `nrfutil`_ and `mcumgr`_ first, and make sure MCUboot's ``imgtool`` is
+available for signing your binary for MCUboot as described on :ref:`west-sign`.
+
+Next, do the **one-time setup** to flash MCUboot. We'll assume you've cloned
+the `MCUboot`_ repository into the directory ``mcuboot``, and that it is next
+to the zephyr repository on your computer.
+
+#. Reset the board into the Nordic bootloader as described above.
```

Comment:
Same here.  Should you wait to reset the board until after you have the software compiled and packaged?

Reply:
MCUboot will stay in serial recovery mode forever.